pkg/pkgmgr/dpkg.go

```diff
@@ -459,115 +464,243 @@ func (dm *dpkgManager) unpackAndMergeUpdates(ctx context.Context, updates unvers
 			})).Root()
 	}
 
+	updated = updated.Run(buildkit.Sh("mkdir -p /tmp/debian-rootfs/var/lib/dpkg")).Root()
+
+	// Replace status file in tooling image with new status file with relevant pacakges from image to be patched.
+	// Regenerate /var/lib/dpkg/info files based on relevant pacakges from image to be patched.
+	dpkgdb := updated.Run(
+		llb.AddEnv("PACKAGES_PRESENT_ALL", string(jsonPackageData)),
+		llb.AddEnv("STATUS_FILE", dm.tempStatusFile),
+		llb.Args([]string{
+			`bash`, `-xec`, `
+							set -ex
+
+							json_str=$PACKAGES_PRESENT_ALL
+
+							rm -r /var/lib/dpkg/info
+							mkdir -p /var/lib/dpkg/info
+
+							apt-get clean
+							apt-get update
+
+							while IFS=':' read -r package version; do
+								pkg_name=$(echo "$package" | sed 's/^"\(.*\)"$/\1/')
+								apt-get install --reinstall -y $pkg_name
+							done <<< "$(echo "$json_str" | tr -d '{}\n' | tr ',' '\n')"
+
+							apt --fix-broken install
+							dpkg --configure -a
+							apt-get check
+
+							echo "$STATUS_FILE" > /var/lib/dpkg/status
+						`,
+		})).Root()
+
 	// Download all requested update packages without specifying the version. This works around:
 	//  - Reports being slightly out of date, where a newer security revision has displaced the one specified leading to not found errors.
 	//  - Reports not specifying version epochs correct (e.g. bsdutils=2.36.1-8+deb11u1 instead of with epoch as 1:2.36.1-8+dev11u1)
 	var downloadCmd string
 	pkgStrings := []string{}
 	if updates != nil {
-		aptGetDownloadTemplate := "apt-get download --no-install-recommends %s"
+		aptGetDownloadTemplate := `
+		if [ "$IGNORE_ERRORS" = "true" ]; then
```

Comment:
Would probably be helpful to move this script into a an embedded file. (see `embed` package https://golang.org/pkg/embed).